tests/templates/test-head-management.php

```diff
@@ -0,0 +1,337 @@
+				'post_excerpt'   => 'Test caption text.',
+			]
+		);
+	}
+
+	/**
+	 * Setup.
+	 */
+	public function setUp() {
+		parent::setup();
+
+		// Disable date organized uploads.
+		add_filter( 'upload_dir', [ $this, 'upload_dir_no_subdir' ] );
+
+		// Set the site icon.
+		update_option( 'site_icon', $this->get_attachment_id() );
+	}
+
+	/**
+	 * Helper used with the `upload_dir` filter to remove the /year/month sub directories from the uploads path and URL.
+	 *
+	 * Taken from the WP PHPUnit test helpers.
+	 *
+	 * @param array $uploads The uploads path data.
+	 * @return array The altered array.
+	 */
+	public function upload_dir_no_subdir( $uploads ) {
+		$subdir = $uploads['subdir'];
+
+		$uploads['subdir'] = '';
+		$uploads['path']   = str_replace( $subdir, '', $uploads['path'] );
+		$uploads['url']    = str_replace( $subdir, '', $uploads['url'] );
+
+		return $uploads;
+	}
+
+	/**
+	 * Return an array that matches the schema for the Components API.
+	 *
+	 * @return array
+	 */
+	public function get_blank_data_array(): array {
+		return [
+			'defaults'       => [],
+			'page'           => [],
+			'providers'      => [],
+			'redirectTo'     => '',
+			'redirectStatus' => 0,
+		];
+	}
+
+	/**
+	 * Return the result from a clean run of `setup_head()`.
+	 *
+	 * @return array
+	 */
+	public function get_head_setup_result(): array {
+		return setup_head(
+			$this->get_blank_data_array(),
+			new \WP_Query(),
+			'site'
+		);
+	}
+
+	/**
+	 * Test that the <head> setup returns the correct component in both the
+	 * `defaults` and `page` component arrays.
+	 */
+	public function test_default_head() {
+
+		// Run `setup_head()` on empty/blank data.
+		$component_endpoint_result = $this->get_head_setup_result();
+
+		// `defaults` should have a <head> component.
+		$this->assertEquals(
+			new Component(
+				'irving/head',
+				[
+					'config' => [
+						'context' => 'defaults',
+					],
+				]
+			),
+			$component_endpoint_result['defaults'][0],
+			'First element in the `defaults` component array is not `irving/head`.'
+		);
+
+		$this->assertEquals(
+			new Component(
+				'irving/head',
+				[
+					'config' => [
+						'context' => 'page',
+					],
+				]
+			),
+			$component_endpoint_result['page'][0],
+			'First element in the `page` component array is not `irving/head`.'
+		);
+	}
+
+	/**
+	 * Test disabling the automated `irving/head` management.
+	 */
+	public function test_disabling_head() {
+
+		// Disable functionality.
+		add_filter( 'wp_irving_setup_head', '__return_false' );
+
+		// Run `setup_head()` on empty/blank data.
+		$component_endpoint_result = $this->get_head_setup_result();
+
+		// Nothing should have happened.
+		$this->assertEquals( [], $component_endpoint_result['defaults'], 'Defaults array was not empty.' );
+		$this->assertEquals( [], $component_endpoint_result['page'], 'Page array was not empty.' );
+
+		// Re-enable functionality.
+		add_filter( 'wp_irving_setup_head', '__return_true' );
```

Comment:
I'm pretty sure this is unnecessary and the WP tests automatically reset hooks after each test, but you may want to confirm. 🔢